### ISSUE TITLE
[SCREENREADER] Add semantic HTML for search results list

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#5555]
## User Story
As an assistive device user, I want the list of search results to be presented clearly so that I can understand them. 

## Task
- Add semantic HTML to list of search results (see additional suggestions on [#7285](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7285)

## Acceptance Criteria
- As a screen reader user I want to have the search results read in with semantic context, so that I can digest the information.
- As a screen reader user I want to hear the introductory sentence in context of the list results.

### Current code
```html
<div class="columns usa-width-one-third medium-4 small-12" id="searchResultsContainer" style="max-height: 78vh; overflow-y: auto;">
<div aria-live="polite" aria-relevant="additions text" class="facility-search-results">
<div>
<div>
<div>
<div class="facility-result" id="vha_402HC">
  <div>
    <p><span><strong>0.1 miles</strong></span></p><span>
    <p><span><span>VA HEALTH</span></span></p>
    <h2 class="vads-u-font-size--h5 no-marg-top">
        <a href="/find-locations/facility/vha_402HC">Portland VA Clinic</a>
    </h2>
    </span>
    <p><span>144 Fore Street<br>Portland, ME 04101-4843</span></p>
  </div>
  <span><a href="https://maps.google.com?saddr=Current+Location&amp;daddr=144 Fore Street, Portland, ME 04101-4843" rel="noopener noreferrer" target="_blank" style="text-decoration: underline;">Directions</a></span>
  <div><div><strong>Main Number: </strong>207-771-3500 <br></div><div><strong>Mental Health: </strong>207-771-3500 x3501<br></div></div>
</div>

<div class="facility-result" id="vha_402GD"><div><p><span><strong>13.5 miles</strong></span></p><span><p><span><span>VA HEALTH</span></span></p><h2 class="vads-u-font-size--h5 no-marg-top"><a href="/find-locations/facility/vha_402GD">Saco VA Clinic</a></h2></span><p><span>655 Main Street<br>Saco, ME 04072-1543</span></p></div><span><a href="https://maps.google.com?saddr=Current+Location&amp;daddr=655 Main Street, Saco, ME 04072-1543" rel="noopener noreferrer" target="_blank" style="text-decoration: underline;">Directions</a></span><div><div><strong>Main Number: </strong>207-294-3100 <br></div><div><strong>Mental Health: </strong>207-294-3100 x3100<br></div></div></div>
<!-- etcetera -->
```

### Suggestion

The code below would read the list contents contextually, as items in an ordered list. The `<dfn>` can be screen reader only so that it is clear what the content is without the visual cues. The `aria-describedby` attribute connects the `<h2>` with the `<ol>`.

```html

<h2 aria-describedby="searchResultsContainer">6 results for VA health near “Portland, Maine 04101, United States”</h2>

<ol class="columns usa-width-one-third medium-4 small-12 facility-search-results"  id="searchResultsContainer" style="max-height: 78vh; overflow-y: auto;" aria-live="polite" aria-relevant="additions text">

  <li class="facility-result" id="vha_402HC">

    <dl>
      <dt><dfn>Distance from your location:</dfn> 0.1 miles
          <dfn>Type of facility:</dfn> VA HEALTH
          <a href="/find-locations/facility/vha_402HC" class="vads-u-font-size--h5 no-marg-top">Portland VA Clinic</a>
      </dt>
      <dd><dfn>Street address:</dfn>144 Fore Street
           <dfn>City:</dfn> Portland, <dfn>State:</dfn> ME 
           <dfn>Zip Code:</dfn> 04101-4843
      </dd>
      <dd>
           <a href="https://maps.google.com?saddr=Current+Location&amp;daddr=144 Fore Street, Portland, ME 04101-4843" rel="noopener noreferrer" target="_blank">Directions <span class="sr-only">to facility</span></a>
      </dd>
      <dd>
           <dfn>Main Number:</dfn> 207-771-3500 
           <dfn><span class="sr-only">Phone number for</span> Mental Health: </dfn>207-771-3500 <abbr title="extension">x</abbr>3501
      </dd>
    </dl>

  </li>

  <li class="facility-result" id="vha_402GD">
<!-- etcetera -->
  </li>
</ol>
```

**Comment by mmiddaugh** [comment]
@jenstrickland will you please fill in my gaps, especially for Acceptance Criteria? Many thanks.

**Comment by mmiddaugh** [comment]
@brangi to review before estimating
This issue is related to the work for correlating icons between map and list

**Comment by mmiddaugh** [comment]
Related to [#5620 ] 
@brangi to determine best order to make these changes

**Comment by mmiddaugh** [comment]
Try to bring into same sprint as [#5620]

**Comment by mmiddaugh** [comment]
I believe this one is a duplicate for #7285 so adding the 508-defect-2 label to this one.

**Comment by mmiddaugh** [comment]
Semantic markup will be very difficult on this current implementation. A design iteration is forthcoming so I will return this to the backlog until then.

**Comment by jenstrickland** [comment]
We've patched the code with some aria and screen-reader only content that should be fine. The design system prevented using semantic markup in this instance.

**Comment by mmiddaugh** [comment]
@jenstrickland I'm looking at this one again as part of my re-review of our priority 508 defects. I note your comment above: 

> We've patched the code with some aria and screen-reader only content that should be fine. 

Does that mean this issue is acceptably resolved?

**Comment by jenstrickland** [comment]
I think it's as good as we're going to get with the current design and development constraints. :(
